internal/controllers/extension_controller_test.go

```diff
@@ -92,6 +86,931 @@ func TestExtensionReconcile(t *testing.T) {
 	}
 }
 
+func TestExtensionNonExistentPackage(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, features.OperatorControllerFeatureGate, features.EnableExtensionAPI, true)()
```

Comment:
These new tests looked like they had similar enough structure to potentially be done in [this kind of format](https://github.com/operator-framework/operator-controller/blob/38da6fc3c70c60d4da570333a3e5fc005192d44f/internal/catalogmetadata/types_test.go#L18). Seems to me like that might be more maintainable?

Reply:
Oops, just realized this is a duplicate comment of Joe's [here](https://github.com/operator-framework/operator-controller/pull/690#pullrequestreview-1931627893).